### ISSUE TITLE
Initialize Mask R-CNN from COCO weights rather than an ImageNet backbone alone

### ISSUE THREAD

**Issue by abuzarmahmood** [pulakatlab/autoslide#121]
Part of #116.

## The defect

`initialize_model()` constructs the detector with no arguments:

https://github.com/pulakatlab/autoslide/blob/main/autoslide/src/pipeline/model/prediction_utils.py#L42

```python
model = torchvision.models.detection.maskrcnn_resnet50_fpn()
```

The resolved defaults are `weights=None`,
`weights_backbone=ResNet50_Weights.IMAGENET1K_V1`. So the ResNet-50 backbone is
pretrained, but the **RPN, the box head and the mask head are all randomly
initialized** — and they are being trained from scratch on 407 images.

## Suggested change

```python
from torchvision.models.detection import (
    maskrcnn_resnet50_fpn, MaskRCNN_ResNet50_FPN_Weights)

model = maskrcnn_resnet50_fpn(weights=MaskRCNN_ResNet50_FPN_Weights.COCO_V1)
# then replace the predictors for num_classes=2, as now
```

COCO initialization gives the detection heads a sensible starting point, which
matters disproportionately at this dataset size. This is likely the single
largest architectural gain available.

## Two further changes worth benchmarking in the same pass

**`maskrcnn_resnet50_fpn_v2`** — improved box/mask heads and a better training
recipe. Drop-in replacement, typically worth a point or two of mask AP.

**Higher mask-head resolution.** The default `mask_roi_pool` output is 14x14,
producing 28x28 masks that are then upsampled to the box. Raising it to 28
(56x56 masks) targets boundary precision specifically, which is what the Dice
metric measures:

```python
model.roi_heads.mask_roi_pool = torchvision.ops.MultiScaleRoIAlign(
    featmap_names=['0', '1', '2', '3'], output_size=28, sampling_ratio=2)
```

Note that COCO weights and a changed `mask_roi_pool` interact — the pretrained
mask head expects 14x14 input — so benchmark COCO-init alone first, then the
resolution change on top.

## Not worth changing

Keep `trainable_backbone_layers=3` and `FrozenBatchNorm2d`. At batch size 2,
frozen BN is correct, and unfreezing more of the backbone on 407 images invites
overfitting.
